pkg/matcher/annotation_matcher.go

```diff
@@ -226,6 +228,14 @@ func MatchPipelinerunByAnnotation(ctx context.Context, logger *zap.SugaredLogger
 		if event.EventType == opscomments.NoOpsCommentEventType.String() || event.EventType == opscomments.OnCommentEventType.String() {
 			continue
 		}
+
+		// logger.Warnf("Warning: The PipelineRun %s has both `on-cel-expression` and `on-event`/`on-target-branch` annotations. The `on-cel-expression` will take precedence, and the other annotations will be ignored.", prun.Name)
+		msg := fmt.Sprintf("Warning: The PipelineRun '%s' has both `on-cel-expression` and `on-event`/`on-target-branch` annotations. The `on-cel-expression` will take precedence, and the other annotations will be ignored.", prun.Name)
+		if prun.GetObjectMeta().GetAnnotations()[keys.OnCelExpression] != "" && (prun.GetObjectMeta().GetAnnotations()[keys.OnEvent] != "" || prun.GetObjectMeta().GetAnnotations()[keys.OnTargetBranch] != "") {
+			logger.Warnf(msg)
+			eventEmitter.EmitMessage(repo, zap.WarnLevel, "RepositoryCannotLocatePipelineRun", msg)
+		}
+
```

Comment:
```suggestion
		msg := fmt.Sprintf("Warning: The PipelineRun '%s' has both `on-cel-expression` and `on-event`/`on-target-branch` annotations. The `on-cel-expression` will take precedence, and the other annotations will be ignored.", prun.Name)
		if prun.GetObjectMeta().GetAnnotations()[keys.OnCelExpression] != "" && (prun.GetObjectMeta().GetAnnotations()[keys.OnEvent] != "" || prun.GetObjectMeta().GetAnnotations()[keys.OnTargetBranch] != "") {
			logger.Warnf(msg)
			eventEmitter.EmitMessage(repo, zap.WarnLevel, "RepositoryCannotLocatePipelineRun", msg)
		}

```

@PuneetPunamiya I think this should be inside if condition of CEL expression and message shoule be customized based on checking `on-event` and `on-target-branch` annotations, checking both annotation separately and updating message